rq/worker.py

```diff
@@ -607,7 +628,8 @@ def work(self, burst: bool = False, logging_level: str = "INFO", date_format=DEF
                     self.scheduler.start()
 
         self._install_signal_handlers()
-
+        retry_num = 0
+        max_retry = 3
```

Comment:
Let's not do anything with retry logic first, at least in this patch. Let's implement `socket_timeout` first and see how it goes. I want to be extra careful to not introduce any new behavior changes.